### PR TITLE
W-14238367 - bug fix and log details for failure

### PR DESCRIPTION
**Pull request by bairenlong-sfdc**

### CODE REVIEW

File: test-utils/src/main/java/com/force/formula/v2/data/FormulaFieldDefinition.java [hunk 1]
```diff
@@ -92,7 +92,7 @@ public Integer getScale() {
      * @return the field value after converting based on the field's data type
      */
     public Object createObjectWithGivenValue(String fieldValue){
-        if (fieldValue == null || fieldValue.length() == 0) {
+        if (fieldValue == null || (this.dataType.isNumber() && fieldValue.trim().length() == 0)) {
```
Comment (bairenlong-sfdc) [inline test-utils/src/main/java/com/force/formula/v2/data/FormulaFieldDefinition.java:95]: Need to handle empty & spaces value for Number filed

File: test-utils/src/main/java/com/force/formula/v2/data/FormulaTestData.java [hunk 2]
```diff
@@ -31,7 +35,7 @@ public class FormulaTestData {
      * @param executionPaths ordered list of execution paths as defined in the test xml file
      * @param referenceFields ordered list of reference fields as defined in the test xml file
      */
-    public FormulaTestData(String input, List<String> outputs, List<String> executionPaths, List<FormulaFieldDefinition> referenceFields) {
+    public FormulaTestData(String testCase, String input, List<String> outputs, List<String> executionPaths, List<FormulaFieldDefinition> referenceFields) {
```
Comment (bairenlong-sfdc) [inline test-utils/src/main/java/com/force/formula/v2/data/FormulaTestData.java:38]: Pass in test case name for logging

File: test-utils/src/main/java/com/force/formula/v2/data/FormulaTestData.java [hunk 3]
```diff
@@ -44,11 +48,28 @@ public FormulaTestData(String input, List<String> outputs, List<String> executio
                 "an expectedOutput in order in which the execution paths are indexed or defined in the testcase");
 
         Map<String, Object> inputMap = new HashMap<>();
+        StringBuffer sb = new StringBuffer();
```
Comment (bairenlong-sfdc) [inline test-utils/src/main/java/com/force/formula/v2/data/FormulaTestData.java:51]: Currently, it is impossible to know what the * is going on on error, it simple drop dead. 
Adding code to log all the details on failure.

File: test-utils/src/main/java/com/force/formula/v2/impl/FormulaTestDefinitionFileParser.java [hunk 4]
```diff
@@ -165,7 +172,7 @@ private List<FormulaTestData> extractTestData(Element element, List<String> exec
                         .map(s -> s.replaceAll("\\\\,", ","))
                         .map(String::trim)
                         .collect(Collectors.toList());
-                testDataList.add(new FormulaTestData(input, outputs, executionPaths, referenceFields));
+                testDataList.add(new FormulaTestData(testCase, input, outputs, executionPaths, referenceFields));
```
Comment (bairenlong-sfdc) [inline test-utils/src/main/java/com/force/formula/v2/impl/FormulaTestDefinitionFileParser.java:175]: Pass in test case name for logging.

**Review by jonathonmar — APPROVED** [review]

**Pull request merged**